File: enemy/role/orc.js
import { Enemy } from '../../enemy/enemy.js';

export class Orc extends Enemy {
    constructor(name, health, maxHealth, potions, strength) {
        super(name, health, maxHealth, potions, strength);
        this.name = name;
        this.health = health;
        this.maxHealth = maxHealth;
        this.potions = potions;
        this.strength = strength;
    }
}

File: Battle-Jawn/src/main/java/com/battlejawn/Entities/Enemy/Orc.java
package com.battlejawn.Entities.Enemy;

import java.time.LocalDateTime;
import jakarta.persistence.*;
import lombok.NoArgsConstructor;

@Entity
@NoArgsConstructor
@DiscriminatorValue("ORC")
public class Orc extends Enemy {

    public Orc(int level, int health, int potions, int strength) {
        super("Orc", health, health, potions, potions, strength, LocalDateTime.now(), level);
    }

}
